.travis.yml

```diff
@@ -2,38 +2,69 @@ sudo: required
 branches:
     only:
         - master
-cache: pip
+cache:
+    pip: True
+    directories:
+      - node_modules
+      - downloads
 language: python
 python:
     - "2.7"
 services:
     - docker
     - memcached
+    - mysql
 env:
+    global:
+        - DATABASE_URL=mysql://root:@127.0.0.1:3306/kuma
+        - DJANGO_SETTINGS_MODULE=kuma.settings.travis
+        - DOCKER_COMPOSE_VERSION=1.9.0
+          # ES_VERSION must come before ES_DOWNLOAD_URL
+        - ES_VERSION=2.4.5
+        - ES_DOWNLOAD_URL=https://download.elastic.co/elasticsearch/release/org/elasticsearch/distribution/tar/elasticsearch/$ES_VERSION/elasticsearch-$ES_VERSION.tar.gz
+        - PIPELINE_CSSMIN_BINARY=$TRAVIS_BUILD_DIR/node_modules/.bin/cssmin
+        - PIPELINE_CSS_COMPRESSOR=pipeline.compressors.cssmin.CSSMinCompressor
+        - PIPELINE_JS_COMPRESSOR=pipeline.compressors.uglifyjs.UglifyJSCompressor
+        - PIPELINE_SASS_BINARY=$TRAVIS_BUILD_DIR/node_modules/.bin/node-sass
+        - PIPELINE_UGLIFYJS_BINARY=$TRAVIS_BUILD_DIR/node_modules/.bin/uglifyjs
     matrix:
         - TOXENV=py27
+          CREATE_DB=kuma
+          INSTALL_PIPELINE=1
+          INSTALL_ELASTICSEARCH=1
+          PYTHONHASHSEED=0
         - TOXENV=flake8
         - TOXENV=docs
-        - TOXENV=locales DOCKER_COMPOSE_VERSION=1.9.0
-        - TOXENV=docker DOCKER_COMPOSE_VERSION=1.9.0 UID=0
+        - TOXENV=locales
+          INSTALL_DOCKER_COMPOSE=1
+        - TOXENV=docker
+          INSTALL_DOCKER_COMPOSE=1
+          UID=0
         - TOXENV=stylelint
-    global:
-        - DJANGO_SETTINGS_MODULE=kuma.settings.travis
-        - DEBIAN_FRONTEND=noninteractive
-        - DATABASE_URL=mysql://root:kuma@localhost:3306/kuma
-        - CFLAGS=-O0
+        # TODO: make this default when http://bugzil.la/1127798 is fixed
+        - TOXENV=py27
+          CREATE_DB=kuma
+          INSTALL_PIPELINE=1
+          INSTALL_ELASTICSEARCH=1
+
 matrix:
     allow_failures:
-        - env: TOXENV=locales DOCKER_COMPOSE_VERSION=1.9.0
-before_install:
-    - scripts/travis-install
-    - pip install -U pip
+        - env: TOXENV=locales INSTALL_DOCKER_COMPOSE=1
+        - env: TOXENV=py27 CREATE_DB=kuma INSTALL_PIPELINE=1 INSTALL_ELASTICSEARCH=1
 install:
-    - pip install -r requirements/travis.txt
-    - if [[ $TOXENV == 'py27' ]]; then ansible-playbook -vvv --tags=mysql,pipeline --connection=local --inventory-file=provisioning/inventory provisioning/travis.yml; fi
     - nvm install 6
```

Comment:
This might be a good follow-on PR.  ``requirements/travis.txt`` includes ``coveralls``, which is called at the end of each test, so you'd need to ensure ``coveralls`` is not called for the targets that generate coverage data.  Which maybe should include the docker target, but currently doesn't.  But it could be skipped for the linter tasks.

That's also a lot of work to save at most 10s per build.